src/main/java/seedu/address/model/property/Property.java

```diff
@@ -31,18 +38,31 @@ public Unit getUnit() {
         return unit;
     }
 
+    public Type getType() {
+        return type;
+    }
+
     /**
-     * Returns true if both properties have the same unit number and postal code.
-     * This defines a weaker notion of equality between two persons.
+     * Returns true if either property is landed and have same postal code
+     * OR is not landed and different type and have same postal code
+     * OR is not landed and same type and same postal code and unit number
+     * This defines a weaker notion of equality between two properties.
      */
     public boolean isSameProperty(Property otherProperty) {
         if (otherProperty == this) {
             return true;
+        } else if (otherProperty == null) {
+            return false;
         }
 
-        return otherProperty != null
-                && otherProperty.getPostalCode().equals(getPostalCode())
+        boolean hasLanded = this.type.equals(new Type(PropertyType.LANDED.toString()));
+
+        if (hasLanded || !otherProperty.getType().equals(getType())) {
+            return otherProperty.getPostalCode().equals(getPostalCode());
+        } else {
```

Comment:
will be good to include a short comment here to explain why the check for landed property is different from other property types